nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/QueryElasticsearchHttp.java

```diff
@@ -0,0 +1,410 @@
+        + "written to Flow Files for transfer.  Also note that the Elasticsearch max_result_window index "
+        + "setting is the upper bound on the number of records that can be retrieved using this query.  "
+        + "To retrieve more records, use the ScrollElasticsearchHttp processor.")
+@WritesAttributes({
+        @WritesAttribute(attribute = "filename", description = "The filename attribute is set to the document identifier"),
+        @WritesAttribute(attribute = "es.index", description = "The Elasticsearch index containing the document"),
+        @WritesAttribute(attribute = "es.type", description = "The Elasticsearch document type"),
+        @WritesAttribute(attribute = "es.result.*", description = "If Target is 'Flow file attributes', the JSON attributes of "
+                + "each result will be placed into corresponding attributes with this prefix.") })
+public class QueryElasticsearchHttp extends AbstractElasticsearchHttpProcessor {
+
+    private static final String FIELD_INCLUDE_QUERY_PARAM = "_source_include";
+    private static final String QUERY_QUERY_PARAM = "q";
+    private static final String SORT_QUERY_PARAM = "sort";
+    private static final String FROM_QUERY_PARAM = "from";
+    private static final String SIZE_QUERY_PARAM = "size";
+
+    public static final String TARGET_FLOW_FILE_CONTENT = "Flow file content";
+    public static final String TARGET_FLOW_FILE_ATTRIBUTES = "Flow file attributes";
+    private static final String ATTRIBUTE_PREFIX = "es.result.";
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description(
+                    "All FlowFiles that are read from Elasticsearch are routed to this relationship.")
+            .build();
+
+    public static final Relationship REL_FAILURE = new Relationship.Builder()
+            .name("failure")
+            .description(
+                    "All FlowFiles that cannot be read from Elasticsearch are routed to this relationship. Note that only incoming "
+                            + "flow files will be routed to failure.").build();
+
+    public static final Relationship REL_RETRY = new Relationship.Builder()
+            .name("retry")
+            .description(
+                    "A FlowFile is routed to this relationship if the document cannot be fetched but attempting the operation again may "
+                            + "succeed. Note that if the processor has no incoming connections, flow files may still be sent to this relationship "
+                            + "based on the processor properties and the results of the fetch operation.")
+            .build();
+
+    public static final PropertyDescriptor QUERY = new PropertyDescriptor.Builder()
+            .name("query-es-query").displayName("Query")
+            .description("The Lucene-style query to run against ElasticSearch").required(true)
```

Comment:
Might be good to have a trivial example of a query here (see my comment on the other processor below)